File: application/core/src/main/java/br/com/beblue/bluerecords/core/repositorio/VendaRepositorio.java
package br.com.beblue.bluerecords.core.repositorio;

import br.com.beblue.bluerecords.core.command.ConsultaVendaCommand;
import br.com.beblue.bluerecords.core.command.RegistraVendaCommand;
import br.com.beblue.bluerecords.core.entitidade.Venda;
import br.com.beblue.bluerecords.core.entitidade.paginacao.Paginacao;

public interface VendaRepositorio {
    Venda cadastrar(RegistraVendaCommand command);
    Paginacao<Venda> consultaVendas(ConsultaVendaCommand command);
    Venda consulta(Integer id);
}

File: application/core/src/main/kotlin/br/com/beblue/bluerecords/core/repositorio/VendaRepositorio.kt
package br.com.beblue.bluerecords.core.repositorio

import br.com.beblue.bluerecords.core.entity.Disco
import br.com.beblue.bluerecords.core.usecase.disco.command.CadastrarDiscoCommand

interface VendaRepositorio {
    fun cadastrar(command: CadastrarDiscoCommand): Disco
}